src/index.ts

```diff
@@ -478,6 +484,22 @@ const handleFilterOperator = (
 		}
 		case '$count': {
 			let keys = ['$count'];
+			if (
+				parentKey != null &&
+				isObject(filter) &&
+				(Object.keys(filter).length === 0 || filter.hasOwnProperty('$filter'))
+			) {
+				keys = parentKey
+					.slice(0, parentKey.length - 1)
+					.concat(
+						handleOptions(
+							'$filter',
+							{ $count: filter as { $filter?: Filter } },
+							parentKey[parentKey.length - 1],
+						),
+					);
```

Comment:
I'd prefer to separate this out to avoid the concat, since the concat both creates another array and also `concat`ing a string is kinda confusing imo
```suggestion
				keys = parentKey.slice(0, parentKey.length - 1);
				keys.push(
					handleOptions(
						'$filter',
						{ $count: filter as { $filter?: Filter } },
						parentKey[parentKey.length - 1],
					),
				);
```